integration-tests/smoke/ccip_batching_test.go

```diff
@@ -0,0 +1,299 @@
+
+	const (
+		numMessages = 5
+	)
+	var (
+		startSeqNum map[uint64]ccipocr3.SeqNum = map[uint64]ccipocr3.SeqNum{
+			sourceChain1: 1,
+			sourceChain2: 1,
+		}
+		endSeqNum map[uint64]ccipocr3.SeqNum = map[uint64]ccipocr3.SeqNum{
+			sourceChain1: ccipocr3.SeqNum(numMessages),
+			sourceChain2: ccipocr3.SeqNum(numMessages),
+		}
+	)
+
+	t.Run("batch data only messages from single source", func(t *testing.T) {
+		sendMessages(
+			ctx,
+			t,
+			e.Env.Chains[sourceChain1],
+			e.Env.Chains[sourceChain1].DeployerKey,
+			state.Chains[sourceChain1].OnRamp,
+			state.Chains[sourceChain1].Router,
+			state.Chains[sourceChain1].Multicall3,
+			destChain,
+			numMessages,
+			common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+		)
+
+		_, err := ccdeploy.ConfirmCommitWithExpectedSeqNumRange(
+			t,
+			e.Env.Chains[sourceChain1],
+			e.Env.Chains[destChain],
+			state.Chains[destChain].OffRamp,
+			nil,
+			ccipocr3.NewSeqNumRange(startSeqNum[sourceChain1], endSeqNum[sourceChain1]),
+		)
+		require.NoErrorf(t, err, "failed to confirm commit from chain %d", sourceChain1)
+
+		startSeqNum[sourceChain1] = endSeqNum[sourceChain1] + 1
+		endSeqNum[sourceChain1] = startSeqNum[sourceChain1] + ccipocr3.SeqNum(numMessages) - 1
+	})
+
+	t.Run("batch data only messages from multiple sources", func(t *testing.T) {
+		var wg sync.WaitGroup
+
+		wg.Add(1)
+		go func(sourceChainSelector uint64) {
+			defer wg.Done()
+			sendMessages(
+				ctx,
+				t,
+				e.Env.Chains[sourceChainSelector],
+				e.Env.Chains[sourceChainSelector].DeployerKey,
+				state.Chains[sourceChainSelector].OnRamp,
+				state.Chains[sourceChainSelector].Router,
+				state.Chains[sourceChainSelector].Multicall3,
+				destChain,
+				numMessages,
+				common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+			)
+		}(sourceChain1)
+
+		wg.Add(1)
+		go func(sourceChainSelector uint64) {
+			defer wg.Done()
+			sendMessages(
+				ctx,
+				t,
+				e.Env.Chains[sourceChainSelector],
+				e.Env.Chains[sourceChainSelector].DeployerKey,
+				state.Chains[sourceChainSelector].OnRamp,
+				state.Chains[sourceChainSelector].Router,
+				state.Chains[sourceChainSelector].Multicall3,
+				destChain,
+				numMessages,
+				common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+			)
+		}(sourceChain2)
+
+		wg.Wait()
+
+		// confirm the commit reports
+		var (
+			sourceChain1Report *offramp.OffRampCommitReportAccepted
+			sourceChain2Report *offramp.OffRampCommitReportAccepted
+		)
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			var err error
+			sourceChain1Report, err = ccdeploy.ConfirmCommitWithExpectedSeqNumRange(t,
+				e.Env.Chains[sourceChain1],
+				e.Env.Chains[destChain],
+				state.Chains[destChain].OffRamp,
+				nil,
+				ccipocr3.NewSeqNumRange(startSeqNum[sourceChain1], endSeqNum[sourceChain1]),
+			)
+			require.NoErrorf(t, err, "failed to confirm commit from chain %d", sourceChain1)
```

Comment:
please note that a require failure here won't fail the actual test immediately. The test will be reported in failed status but after all go routine is completed.
If you want to fail the test immediately for any failure in individual routines please consider using errgroup.

Reply:
Are you sure? `require` calls `t.FailNow` which should shut down the whole test

Reply:
Fixed